### DISCUSSION TITLE
replicad CLI/desktop app

### DISCUSSION THREAD

**Discussion by dcorbin** [sgenoud/replicad#168] [General]
I have tool that generates openscad, and then uses openscad for previewing the shape, and to generate STL.   I'm contemplating switching to use replicad instead of openscad.  

To do that, I basically need two things:  
1) a command-line tool that can take a replicad model and generate the .STL for it and 
2) a "viewer" that presents view similar to the workbench

I _think_ #1 is doable using node.  Does something like this exist already?  How hard would it be on a scale of 1-10 to create?

 As for #2, I could I think finagle something with the browser, though that seems less ideal to me.  More appealing to me (conceptually) is to have some gui-based desktop app that takes a file on the command line, and renders it it's own GUI window.  The same two questions apply:   Does something like this exist already?  How hard would it be on a scale of 1-10 to create?
 
 Thanks

**Comment by sgenoud** [comment]
For #1 I have create a CLI to generate STL (using Deno and not node): https://github.com/sgenoud/replicad-cli

This is a small project that I mostly built for my purpose (generating project images for [my models log](https://models.sgenoud.com/)), but it should work for STL as well.

**Comment by sgenoud** [comment]
For #2 I have something similar (and actually part of the workbench). If you are using a Chromium based browser (sorry, but the others do not want to support the APIs needed for this), you can click the arrow on the left (toggle auto-reload) or just go [here](https://studio.replicad.xyz/visualiser).

I am trying to make the whole thing work offline as well (there are some bugs I need to fix though). You could try to wrap the whole thing in an electron shell to make it callable from the command line.
